packages/tex-to-myst/src/algorithms.ts

```diff
@@ -0,0 +1,172 @@
+import { u } from 'unist-builder';
+import type { GenericNode } from 'myst-common';
+import { normalizeLabel } from 'myst-common';
+import { type Paragraph } from 'myst-spec';
+import { type Line } from 'myst-spec-ext';
+import type { Handler, ITexParser } from './types.js';
+import { getArguments } from './utils.js';
+import { select, selectAll } from 'unist-util-select';
+
+function addNestingStatement(
+  node: GenericNode,
+  state: ITexParser,
+  { before, after }: { before: string; after?: string },
+) {
+  state.closeParagraph();
+  state.openParagraph({ indent: state.data.algorithm_indent });
+  state.pushNode(u('strong', [u('text', before)]));
+  const args = getArguments(node, 'group');
+  const children = args[args.length - 1];
+  if (children) {
+    state.text(' ');
+    state.renderChildren(children);
+  }
+  if (after) {
+    state.text(' ');
+    state.pushNode(u('strong', [u('text', after)]));
+  }
+  state.closeParagraph();
+  state.data.algorithm_indent ??= 0;
+  state.data.algorithm_indent += 1;
+}
+
+function finishNestingStatement(node: GenericNode, state: ITexParser, { text }: { text: string }) {
+  state.closeParagraph();
+  state.data.algorithm_indent ??= 0;
+  state.data.algorithm_indent -= 1;
+  state.openParagraph({ indent: state.data.algorithm_indent });
+  state.pushNode(u('strong', [u('text', text)]));
+  const args = getArguments(node, 'group');
+  const children = args[args.length - 1];
+  if (children) {
+    state.text(' ');
+    state.renderChildren(children);
+  }
+  state.closeParagraph();
+}
+
+function numberParagraphsAsLines(node: GenericNode) {
+  const paragraphs = selectAll('paragraph', node) as Paragraph[];
+  paragraphs.forEach((p, i) => {
+    const l = p as unknown as Line;
+    l.type = 'line';
+    l.enumerator = String(i + 1);
```

Comment:
This enumeration is at a granular level, without the customization allowed for other enumerators. Again, probably fine but it means the `enumerator` attribute works in different ways.

Maybe, though, having all the normal enumerator customization here is also the long-term goal.

Reply:
I did start out with this being called "number", but also felt like we should use common terminology. Certainly a bit different than the other enumerators, with this one being much more granular.